workflow/python/ds-pack/v1/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.8.13-slim-buster
+FROM python:3.8-slim-buster
```

Comment:
Hey @arjun8115 - Guess we could have updated it to a higher version ? Something like `python:3.10.10-slim` which could have further reduced the vulnerabilities ? This applies to all `Dockerfile` changes in this PR.
<img width="1142" alt="image" src="https://user-images.githubusercontent.com/86290535/233654721-2dfa0f84-380b-4f5e-88a5-7cbd063db5bf.png">

Reply:
Hey @peak-jiteshahuja I have changed it to a major version which already has much lesser vulnerabilities (checked locally). Also, `slim` contains less no. of packages, and upgrading to a higher version might break the flow. That's why I kept it as same major version.

Reply:
Hey although i do understand your point here, i'd suggest trying with the `3.10` with some existing/new workspaces/workflows and checking with DS once. Moreover since we would be actually releasing a new version/image after the PR gets merged i think it should be fine since DS would have a choice to either use the old `3.8` image or this new one. Pl Let me know if this makes sense.
cc - @parul157